pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -584,24 +585,56 @@ func newHTTPPassThroughFilterChain(configgen *ConfigGeneratorImpl,
 	return filterChains
 }
 
-func newTCPProxyOutboundListenerFilter(push *model.PushContext, node *model.Proxy) *listener.Filter {
+func buildFallthroughNetworkFilters(push *model.PushContext, node *model.Proxy,
+	l *xdsapi.Listener) []*listener.Filter {
 	tcpProxy := &tcp_proxy.TcpProxy{
 		StatPrefix:       util.BlackHoleCluster,
 		ClusterSpecifier: &tcp_proxy.TcpProxy_Cluster{Cluster: util.BlackHoleCluster},
 	}
+	requireSniForwarding := false
 	if util.IsAllowAnyOutbound(node) {
 		// We need a passthrough filter to fill in the filter stack for orig_dst listener
+		egressCluster := util.PassthroughCluster
+
+		// no need to check for nil value as the previous if check has checked
+		if node.SidecarScope.OutboundTrafficPolicy.EgressProxy != nil {
```

Comment:
can OutboundTrafficPolicy be nil?

Reply:
Yes right? If I dont define it in the sidecar, then it defaults to the ambient defaults.

Reply:
I thought so as well.. won't we panic here if so?

Reply:
The default sidecarscope, it is not nil. There is no cases caught this.

Reply:
`IsAllowAnyOutbound()` is already doing nil check for `OutboundTrafficPolicy`. Should we check again?

Reply:
no need then